test/e2e/garbage_collector.go

```diff
@@ -719,4 +755,52 @@ var _ = framework.KubeDescribe("Garbage collector", func() {
 			framework.Failf("failed to wait for all pods to be deleted: %v", err)
 		}
 	})
+
+	It("should delete jobs and pods created by cronjob", func() {
+		framework.SkipIfMissingResource(f.ClientPool, CronJobGroupVersionResource, f.Namespace.Name)
+
+		By("Create the cronjob")
+		cronJob := newCronJob("simple", "*/1 * * * ?")
+		cronJob, err := f.ClientSet.BatchV2alpha1().CronJobs(f.Namespace.Name).Create(cronJob)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Wait for the CronJob to create new Job")
+		err = wait.PollImmediate(500*time.Millisecond, 2*time.Minute, func() (bool, error) {
+			jobs, err := f.ClientSet.Batch().Jobs(f.Namespace.Name).List(metav1.ListOptions{})
+			if err != nil {
+				return false, fmt.Errorf("Failed to list jobs: %v", err)
+			}
+			return len(jobs.Items) > 0, nil
+		})
+		if err != nil {
+			framework.Failf("Failed to wait for the CronJob to create some Jobs: %v", err)
+		}
+
+		By("Delete the cronjob")
+		if err := f.ClientSet.BatchV2alpha1().CronJobs(f.Namespace.Name).Delete(cronJob.Name, getNonOrphanOptions()); err != nil {
+			framework.Failf("Failed to delete the CronJob: %v", err)
+		}
+		By("Wait for the CronJob to be removed")
+		err = wait.PollImmediate(500*time.Millisecond, 1*time.Minute, func() (bool, error) {
+			_, err := f.ClientSet.BatchV2alpha1().CronJobs(f.Namespace.Name).Get(cronJob.Name, metav1.GetOptions{})
+			return err != nil && !errors.IsNotFound(err), nil
+		})
+		By("Verify if cronjob does not leave jobs nor pods behind")
```

Comment:
These should be wrapped in a poll. The Cronjob is deleted with getNonOrphanOptions, which is equivalent of the "background" GC policy, which causes the cronjob to be deleted immediately, then GC deletes the dependents in the background.